src/engine_openssl.c

```diff
@@ -891,21 +894,30 @@ static int generate_csr(tls_private_key key, char **pem, size_t *pemlen, ...) {
         X509_NAME_add_entry_by_txt(subj, id, MBSTRING_ASC, val, -1, -1, 0);
     }
 
+#define ssl_check(OP) do{ \
+op = #OP;                          \
+if((OP) == 0) {             \
```

Comment:
should this be < 1 just in case one of these functions ever decides to return a negative value to indicate failure? i checked the three that were used already but it seems like it might be a good idea?

Reply:
0 for error is the standard for all OpenSSL functions